src/main/java/redis/clients/jedis/JedisClusterCommand.java

```diff
@@ -117,18 +118,29 @@ private T runWithRetries(byte[] key, int redirections, boolean tryRandomNode, bo
       }
 
       return execute(connection);
-    } catch (JedisConnectionException jce) {
-      if (tryRandomNode) {
-        // maybe all connection is down
-        throw jce;
-      }
 
+    } catch (JedisNoReachableClusterNodeException jnrcne) {
+      //Not sure if JedisNoReachableClusterNodeException exception should extends JedisConnectionException, it's not connection exception
+      //Remove this if hierarchy reworked and no need to make exception from JedisConnectionException clause
+      throw jnrcne;
+    } catch (JedisConnectionException jce) {
       // release current connection before recursion
       releaseConnection(connection);
       connection = null;
 
-      // retry with random connection
-      return runWithRetries(key, redirections - 1, true, asking);
+      if (redirections <= 1) {
```

Comment:
> Now this name is absolutely misleading. It's not number of redirections, it's total number of attempts. So... maxRedirection = 1 means that there are no redirections, only one attempt. I would suggest to rename it to maxAttempts.

This is with the modifications on this PR correct?. Because as JedisCluster is today, if you sex maxRedirections to 1, you can get redirected only once.